File: recyviewdatabinding/src/main/java/mohan/com/recyviewdatabinding/presentation/view/MainActivity.java
package mohan.com.recyviewdatabinding.presentation.view;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import mohan.com.recyviewdatabinding.R;
import mohan.com.recyviewdatabinding.data.model.Employee;
import mohan.com.recyviewdatabinding.databinding.ActivityMainBinding;
import mohan.com.recyviewdatabinding.presentation.adapter.EmployeeDataAdapter;
import mohan.com.recyviewdatabinding.presentation.viewmodel.MainViewModel;

public class MainActivity extends AppCompatActivity {
    private MainViewModel mainViewModel;
    private EmployeeDataAdapter employeeDataAdapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding activityMainBinding =DataBindingUtil.setContentView(this, R.layout.activity_main);

        // bind RecyclerView
        RecyclerView recyclerView = activityMainBinding.viewEmployees;
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setHasFixedSize(true);

        mainViewModel = ViewModelProviders.of(this).get(MainViewModel.class);
        employeeDataAdapter = new EmployeeDataAdapter();
        recyclerView.setAdapter(employeeDataAdapter);
        getAllEmployee();
    }

    private void getAllEmployee() {
        mainViewModel.getAllEmployee().observe(this, new Observer<List<Employee>>() {
            @Override
            public void onChanged(@Nullable List<Employee> employees) {
                employeeDataAdapter.setEmployeeList((ArrayList<Employee>) employees);
            }
        });
    }
}

File: myfoodrecipe/src/main/java/com/example/myrecipe/MainActivity.kt
package com.example.myrecipe

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {

    @SuppressLint("WrongConstant")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        //getting recyclerview from xml
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)

        //adding a layoutmanager
        //recyclerView.layoutManager = LinearLayoutManager(this, LinearLayout.VERTICAL, false)

        recyclerView.layoutManager =
            GridLayoutManager(applicationContext, 2)

        //crating an arraylist to store users using the data class user
        val users = ArrayList<Reciepe>()

        //adding some dummy data to the list
        users.add(
            Reciepe(
                "Neapolitan Pizza",
                "https://cdnimg.webstaurantstore.com/uploads/buying_guide/2014/11/pizzatypes-margherita-.jpg"
            )
        )
        users.add(Reciepe("Chicago Pizza", "http://img.recipepuppy.com/694321.jpg"))
        users.add(Reciepe("New York-Style Pizza", "http://img.recipepuppy.com/707237.jpg"))
        users.add(Reciepe("Sicilian Pizza", "http://img.recipepuppy.com/694321.jpg"))
        users.add(Reciepe("Greek Pizza", "http://img.recipepuppy.com/707237.jpg"))
        users.add(Reciepe("California Pizza", "http://img.recipepuppy.com/694321.jpg"))
        users.add(Reciepe("Detroit Pizza", "http://img.recipepuppy.com/707237.jpg"))
        users.add(Reciepe("St. Louis Pizza", "http://img.recipepuppy.com/694321.jpg"))
        users.add(Reciepe("Types of Pizza Crust", "http://img.recipepuppy.com/707237.jpg"))
        //creating our adapter
        val adapter = CustomAdapter(users)

        //now adding the adapter to recyclerview
        recyclerView.adapter = adapter
    }
}
